pages/entropy/current-fees.mdx

```diff
@@ -26,10 +31,14 @@ Note that the fees shown below will vary over time with prevailing gas prices on
   fees.
 </Callout>
 
-<EntropyFeeTable
-  deployments={Object.fromEntries(
-    Object.entries(EntropyDeployments).filter(
-      ([k, v]) => v.network === "testnet" && v.rpc !== undefined
+<EntropyApiDataFetcher url={FORTUNA_API_URLS.TESTNET}>
+  {(data, isLoading, error) =>
+    isLoading ? (
+      <p>Loading testnet fees...</p>
+    ) : error ? (
+      <p>Error loading testnet fees: {error}</p>
+    ) : (
+      <EntropyFeeTable deployments={data} />
     )
-  )}
-/>
+  }
+</EntropyApiDataFetcher>
```

Comment:
There's a lot of duplication here, what I suggest is after you change `EntropyDataApiFetcher` into a hook, make a component like this to use from your markdown:

```tsx
export const EntropyFees = ({ networkName, url }) => {
  const { isLoading, error, data } = useEntropyApiData(url);
  if (isLoading) {
    return <p>Loading {networkName} fees...</p>
  } else if (error) {
    return <p>Error loading {networkName} fees: {error}</p>
  } else {
    return <EntropyFeeTable deployments={data} />
  }
}

...

<EntropyFees networkName="mainnet" url={FORTUNA_API_URLS.MAINNET} />
```

and then do the same for the contract addresses.

Reply:
I like that.